File: fixtures/coding/maintainability-mini-service-go/container.go
package miniservice

// BuildApp builds the mini application and registers routes.
func BuildApp() *Application {
	app := NewApplication(NewCustomerRepository(), NewAuditLog())
	app.AddRoute("GET", "/customers", ListCustomers)
	app.AddRoute("POST", "/customers", CreateCustomer)
	app.AddRoute("POST", "/customers/bulk-import", BulkImportCustomersHandler)
	return app
}

File: fixtures/coding/maintainability-mini-service-rust/src/container.rs
use crate::audit::AuditLog;
use crate::customers::{bulk_import_customers, create_customer, list_customers};
use crate::repository::CustomerRepository;
use crate::router::Application;

/// Builds the mini application and registers routes.
pub fn build_app() -> Application {
    let mut app = Application::new(CustomerRepository::new(), AuditLog::new());
    app.add_route("GET", "/customers", list_customers);
    app.add_route("POST", "/customers", create_customer);
    app.add_route("POST", "/customers/bulk-import", bulk_import_customers);
    app
}
